Client/WebFilters/ShieldStats/Adblock/AdblockRustEngine.swift

```diff
@@ -0,0 +1,30 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+
+/// A wrapper around adblock_rust_lib header.
+class AdblockRustEngine {
+    private let engine: OpaquePointer
+    
+    init(rules: String = "") { engine = engine_create(rules) }
+    deinit { engine_destroy(engine) }
+    
+    func shouldBlock(requestUrl: String, requestHost: String, sourceHost: String) -> Bool {
+        var explicitCancel = false
+        var savedFromException = false
+        let thirdPartyRequest = requestHost != sourceHost
+        
+        var emptyPointer: UnsafeMutablePointer<Int8>?
+        let redirect = UnsafeMutablePointer(mutating: &emptyPointer)
+        
+        return engine_match(engine, requestUrl, requestHost,
```

Comment:
The `redirect` pointer is passed in but the out value is never used?

Also, do you know if:

`engine_match(...., UnsafeMutablePointer(mutating: &emptyPointer))`

would would instead of creating `redirect`?

Reply:
yes, we don't use any of these inout values but the library requires them, that's why I put this code in a Swift wrapper

I don't understand this part, could you rephrase
```
engine_match(...., UnsafeMutablePointer(mutating: &emptyPointer))
would would instead of creating redirect?
```

I tried passing nil value there but the lib was crashing

In general, this library is used for ad blocking on desktop and Android, while we on iOS use it only for checking if a resource is blocked, that's why most of it is not utilized

Reply:
@iccub 

```swift
func engine_match(..., ..., redirect: UnsafeMutablePointer<UnsafeMutablePointer<Int8>?>)
```

Instead of doing:
```swift
var emptyPointer: UnsafeMutablePointer<Int8>?
let redirect = UnsafeMutablePointer(mutating: &emptyPointer)

engine_match(..., ..., redirect)
```

Would this work:

```swift
var emptyPointer: UnsafeMutablePointer<Int8>?
engine_match(..., ..., UnsafeMutablePointer(mutating: &emptyPointer))
```

Essentially getting rid of `redirect` as a temporary variable. Just curious if it would compile and work.

Reply:
it works and compiles, changing it to your version :)